docs/mirror_configuration.md

```diff
@@ -1,272 +1,408 @@
 
-## compare-method
+This is useful to sync most of the files from an existing, nearby mirror - for example, when creating a new mirror identical to an existing one for the purpose of load sharing.
 
-The compare method is used to set how to compare an existing file with upstream file to determine whether a download is required:
+### `download-mirror-no-fallback`
 
-- hash: this is the default which reads local file content and computes hashes (currently sha256sum), it is reliable but sometimes slower;
-- stat: use file size and change time to compare, which is named after the stat() syscall, this avoids retrieving the full file content thus reducing some io workloads.
+Disable the fallback behavior for [](./mirror_configuration.md#download-mirror).
 
-Example:
+:Type: boolean
+:Default: false
 
-```ini
-[mirror]
-compare-method = hash
+When set to `true`, Bandersnatch only downloads package distribution artifacts from the repository set in [](./mirror_configuration.md#download-mirror) and ignores file URLs received from the [](./mirror_configuration.md#master) server.
+
+```{warning}
+This could lead to more failures than expected and is not recommended for most scenarios.
 ```
 
-## proxy
+%
+% processing and miscellaneous options
+%
 
-The proxy is used only when requesting master server, eg. downloading index or package file from pypi.org.
-The proxy value will be passed to aiohttp as proxy parameter, like `aiohttp.get(link, proxy=yourproxy)`,
-check the aioproxy manual for more details: <https://docs.aiohttp.org/en/stable/client_advanced.html#proxy-support>
+### `cleanup`
 
-Example:
+Enable cleanup of legacy simple directories with non-normalized names.
 
-```ini
-[mirror]
-proxy=http://myproxy.com
+:Type: boolean
+:Default: false
+
+Bandersnatch versions prior to 4.0 used directories with non-normalized package names for compatability with older versions of pip. Enabling this option checks for and removes these directories.
+
+```{seealso}
+[Python Packaging User Guide - Names and Normalization](https://packaging.python.org/en/latest/specifications/name-normalization/)
 ```
 
-## download-mirror
+### `workers`
 
-By default bandersnatch downloads packages from the URL supplied in the master server server's json response.
-This option asks bandersnatch to try to download from the configured PyPI mirror first, and fallback to the
-URL supplied by the master server if it was not successful (unable to get content or checksum mismatch).
-This is useful to sync most of the files from an existing, nearby mirror, for example when setting up a new
-server sitting next to an existing one for the purpose of load sharing.
+The number of worker threads used for parallel downloads.
 
-Example:
+:Type: number, 1 ≤ N ≤ 10
+:Default: 3
 
-```ini
-[mirror]
-download-mirror = https://pypi-mirror.example.com/
-```
+Recommendations:
+
+- leave the default of 3 to avoid overloading the pypi master
```

Comment:
```suggestion
- leave the default of 3 to avoid overloading the pypi master and maybe even your own internet connection
  - if you see timeouts and have a slow connection, try lowering this setting
```

Reply:
Noted, haha